### ISSUE TITLE
symqemu can not run normally on program which call mmap to read file

### ISSUE THREAD

**Issue by hackoflpf** [eurecom-s3/symqemu#30]
i wanto use symqemu to find new paths on freetype-demos ftbench program,but it can't run symbolic execution properly.I have set the environment variables correctly.I use /bin/cat to test,it runs normally.
![image](https://github.com/eurecom-s3/symqemu/assets/23718380/24918fb4-e259-4d24-832b-2b8d8a030762)
However,when i use ftbench program,it doesn't run normally.
![image](https://github.com/eurecom-s3/symqemu/assets/23718380/537785ca-16f5-4cdf-b784-592122bb23ff)
Finally,i try to debug the program,i find it can not call sym_make_symbolic function,because the ftbench program use mmap to read file.

**Comment by aurelf** [comment]
I guess that needs instrumentation.

**Comment by sebastianpoeplau** [comment]
Yes, we would need to intercept the `mmap` call and mark the target memory region as symbolic input bytes.
